### PR TITLE
gh-110260: Check for `PyList_SetItem` errors in `termios` module

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-110260

### CODE REVIEW

**Comment by sobolevn** [comment]
There are other cases with different functions / modules. But, I want to be short in my PR, this way it is easier to merge them :)

File: Modules/termios.c [hunk 1]
```diff
@@ -120,37 +120,51 @@ termios_tcgetattr_impl(PyObject *module, int fd)
...
     if ((mode.c_lflag & ICANON) == 0) {
         v = PyLong_FromLong((long)mode.c_cc[VMIN]);
-        if (v == NULL)
+        if (v == NULL) {
+            goto err;
+        }
+        if (PyList_SetItem(cc, VMIN, v) < 0) {
             goto err;
-        PyList_SetItem(cc, VMIN, v);
+        };
```
Comment (serhiy-storchaka) [inline Modules/termios.c]:
```suggestion
        }
```

File: Modules/termios.c [hunk 2]
```diff
@@ -120,37 +120,51 @@ termios_tcgetattr_impl(PyObject *module, int fd)
...
+    if (!(v = PyList_New(7))) {
         goto err;
     }
-    PyList_SetItem(v, 6, cc);
+
+#define ADD_LONG_ITEM(index, val) \
+    do { \
+        PyObject *l = PyLong_FromLong((long)val); \
+        if (l == NULL) { \
+            goto err; \
```
Comment (serhiy-storchaka) [inline Modules/termios.c:155]: `v` is leaked here.

File: Modules/termios.c [hunk 3]
```diff
@@ -120,37 +120,52 @@ termios_tcgetattr_impl(PyObject *module, int fd)
...
     /* Convert the MIN and TIME slots to integer.  On some systems, the
        MIN and TIME slots are the same as the EOF and EOL slots.  So we
        only do this in noncanonical input mode.  */
     if ((mode.c_lflag & ICANON) == 0) {
         v = PyLong_FromLong((long)mode.c_cc[VMIN]);
-        if (v == NULL)
+        if (v == NULL) {
+            goto err;
+        }
+        if (PyList_SetItem(cc, VMIN, v) < 0) {
```
Comment (erlend-aasland) [inline Modules/termios.c:134]: This can be replaced with `PyList_SET_ITEM`; we know that `cc` is a list.

File: Modules/termios.c [hunk 4]
```diff
@@ -120,37 +120,52 @@ termios_tcgetattr_impl(PyObject *module, int fd)
...
             goto err;
-        PyList_SetItem(cc, VMIN, v);
+        }
         v = PyLong_FromLong((long)mode.c_cc[VTIME]);
-        if (v == NULL)
+        if (v == NULL) {
+            goto err;
+        }
+        if (PyList_SetItem(cc, VTIME, v) < 0) {
```
Comment (erlend-aasland) [inline Modules/termios.c:141]: This can be replaced with `PyList_SET_ITEM`; we know that `cc` is a list.

File: Modules/termios.c (hunk 3 above)
Comment (serhiy-storchaka) [reply Modules/termios.c:134]: But what if VMIN or VTIME are out of range? We do not know this, their definition is out of our control.

Comment (erlend-aasland) [reply Modules/termios.c:134]: Ah, good catch. Disregard my comment.

File: Modules/termios.c (hunk 4 above)
Comment (erlend-aasland) [reply Modules/termios.c:141]: See https://github.com/python/cpython/pull/110261#discussion_r1345477502

**Review by erlend-aasland — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**